scipy/sparse/csgraph/_flow.pyx

```diff
@@ -0,0 +1,395 @@
+    vertex in :math:`U` and a sink vertex with edges from every vertex in
+    :math:`V`. Finally, give every edge in the resulting graph a capacity of 1.
+    Then, a maximum flow in the new graph gives a maximum matching in the
+    original graph consisting of the edges in :math:`E` whose flow is positive.
+
+    Assume that the edges are represented by a
+    :math:`\lvert U \rvert \times \lvert V \rvert` matrix in CSR format whose
+    :math:`(i, j)`'th entry is 1 if there is an edge from :math:`i \in U` to
+    :math:`j \in V` and 0 otherwise; that is, the input is of the form required
+    by :func:`scipy.sparse.csgraph.maximum_bipartite_matching`. Then the CSR
+    representation of the graph constructed above contains this matrix as a
+    block. Here's an example:
+
+    >>> graph = csr_matrix([[0, 1, 0, 1], [1, 0, 1, 0], [0, 1, 1, 0]])
+    >>> print(graph.toarray())
+    [[0 1 0 1]
+     [1 0 1 0]
+     [0 1 1 0]]
+    >>> i, j = graph.shape
+    >>> n = graph.nnz
+    >>> indptr = np.concatenate([[0],
+    ...                          graph.indptr + i,
+    ...                          np.arange(n + i + 1, n + i + j + 1),
+    ...                          [n + i + j]])
+    >>> indices = np.concatenate([np.arange(1, i + 1),
+    ...                           graph.indices + i + 1,
+    ...                           np.repeat(i + j + 1, j)])
+    >>> data = np.ones(n + i + j, dtype=np.int)
+    >>>
+    >>> graph_flow = csr_matrix((data, indices, indptr))
+    >>> print(graph_flow.toarray())
+    [[0 1 1 1 0 0 0 0 0]
+     [0 0 0 0 0 1 0 1 0]
+     [0 0 0 0 1 0 1 0 0]
+     [0 0 0 0 0 1 1 0 0]
+     [0 0 0 0 0 0 0 0 1]
+     [0 0 0 0 0 0 0 0 1]
+     [0 0 0 0 0 0 0 0 1]
+     [0 0 0 0 0 0 0 0 1]
+     [0 0 0 0 0 0 0 0 0]]
+
+    At this point, we can find the maximum flow between the added sink and the
+    added source and the desired matching can be obtained by restricting the
+    residual graph to the block corresponding to the original graph:
+
+    >>> flow = maximum_flow(graph_flow, 0, i+j+1)
+    >>> matching = flow.residual[1:i+1, i+1:i+j+1]
+    >>> print(matching.toarray())
+    [[0 1 0 0]
+     [1 0 0 0]
+     [0 0 1 0]]
+
+    This tells us that the first, second, and third vertex in :math:`U` are
+    matched with the second, first, and third vertex in :math:`V` respectively.
+
+    While this solves the maximum bipartite matching problem in general, note
+    that algorithms specialized to that problem will perform better. In
+    particular, :func:`scipy.sparse.csgraph.maximum_bipartite_matching` will be
+    faster when its preconditions are met.
+
+    This approach can also be used to solve various common generalizations of
+    the maximum bipartite matching problem. If, for instance, some vertices can
+    be matched with more than one other vertex, this may be handled by
+    modifying the capacities of the new graph appropriately.
+
+    References
```

Comment:
References should go above Examples

Reply:
The reference to CLRS ([2]) is used in the Examples section, but I guess that's fine.